tests/python/unittest/test_io.py

```diff
@@ -313,6 +313,20 @@ def check_CSVIter_synthetic():
 
     check_CSVIter_synthetic()
 
+
+def test_DictData_NDIterator():
+    train_iter = mx.io.NDArrayIter({
+        'i0': mx.nd.zeros(5, ),
+        'i1': mx.nd.ones(5, ),
+    }, batch_size=5)
+
+    net = mx.sym.var('i1')
+    mod = mx.module.Module(net, ['i1'], None)
+    mod.bind([('i1', (5,))], None)
+    mod.init_params()
+    mod.predict(train_iter)
+    assert_almost_equal(mod.get_outputs()[0].asnumpy(), np.ones(shape=(5,)))
```

Comment:
Please add more unit tests to increase test coverage, including assert exception.